### PR TITLE
Fix style

### PR DESCRIPTION
**Pull request by shortdudey123**
@brianbianco let me know your thoughts
I can disable specific cops and revert them if you wish

### CODE REVIEW

**Comment by martinb3** [comment]
LGTM. I'd like to merge this, after I run the test-kitchen suites 👍

**Comment by shortdudey123** [comment]
Sounds good :)

**Comment by shortdudey123** [comment]
@martinb3 reverted a change since it broke the spec tests and i missed that

**Comment by martinb3** [comment]
Hello! All of the 'default' test suites are failing with these changes. I'm going to comment on one example of the problem on a specific line, but I haven't chased it in detail. There may be more.

File: providers/install.rb [hunk 1]
```diff
@@ -52,43 +54,63 @@ def download
...
 def unpack
-  install_dir = "#{new_resource.base_name}#{new_resource.version}"
+  install_dir = "#{new_resource.base_name}#{new_resource.version}" # rubocop:disable Lint/UselessAssignment
   case new_resource.artifact_type
-    when "tar.gz",".tgz"
-      execute %(cd #{new_resource.download_dir} ; mkdir -p '#{install_dir}' ; tar zxf '#{@tarball}' --strip-components=1 -C '#{install_dir}' --no-same-owner)
-    else
-      raise Chef::Exceptions::UnsupportedAction, "Current package type #{new_resource.artifact_type} is unsupported"
+  when 'tar.gz', '.tgz'
+    execute %w(
```
Comment (martinb3) [inline providers/install.rb]: It seems that `%w()` is for non-interpolated strings, and so many of these commands are now broken:
```
           ================================================================================
           Error executing action `run` on resource 'redisio_install[redis-installation]'
           ================================================================================

           RuntimeError
           ------------
           cd #{new_resource.download_dir} ; mkdir -p '#{install_dir}' ; tar zxf '#{@tarball}' --strip-components=1 -C '#{install_dir}' --no-same-owner
```

Comment (shortdudey123) [inline providers/install.rb]: Figured that might happen.  Will revert the lines i broke up for `execute`

**Comment by shortdudey123** [comment]
test kitchen passes on all 4 `ubuntu-1404` tests now

**Comment by shortdudey123** [comment]
Passes on all kitchen tests now :)

**Comment by brianbianco** [comment]
Maybe this is just not a valid concern but by switching from :symbol to symbol: in some of the code it actually breaks compatibility with older versions of ruby and isn't just a stylistic / readability change.  Of course 1.8.x was EOL a very long time ago so maybe this is just me being overly cautious or overly accommodating.

**Comment by shortdudey123** [comment]
:symbol vs symbol: was changed in 1.9 and currently only 2.2+ is getting bug fixes with 2.1 getting only security updates (see https://www.ruby-lang.org/en/news/2016/02/24/support-plan-of-ruby-2-0-0-and-2-1/)

Not sure if its still worth trying to support < 1.9

**Comment by brianbianco** [comment]
Ok I'll bite on that.  I am going to go ahead and merge this unless @martinb3 or @zarry object.

**Comment by shortdudey123** [comment]
I can add a note in the readme about only supporting ruby 2+ if you think that will help :)

**Comment by martinb3** [comment]
Works for me now, with tests passing. This includes a lot of commits; I think squashing it (so we don't have any intermediate, broken commits, that someone might checkout and try to use) would be good to keep the repo history clean.

**Comment by shortdudey123** [comment]
All squashed into 1 commit?
or 1 for each: rubocop, test kitchen, and food critic?

**Comment by martinb3** [comment]
I'm thinking just squash the "Revert..." ones into the ones before them, so we don't have any commits that are actually broken / failing tests.

**Comment by shortdudey123** [comment]
Done :)

**Comment by shortdudey123** [comment]
@martinb3 any additional feedback?

**Comment by martinb3** [comment]
@shortdudey123 the only other thing I notice is the use of `:use_inline_resources` -- while I know it's a style/lint check, it could potentially obscure resources from other cookbooks or recipes trying to notify them. I don't think it's a big issue for this cookbook, but could present a breaking a change if we had users notifying a resource declared in our providers, and now they couldn't.

I've had a few branches over the last year or two that fixed lint/style stuff, and I don't have a good answer for this one other than (a) we avoid `:use_inline_resources` or (b) we break compatibility.

**Comment by shortdudey123** [comment]
I see your point, however, i don't see any notifies in the providers which is what would break.

**Comment by brianbianco** [comment]
using use_inline_resources will break compatibility with older versions of chef (thats why redisio has its weird way of handling inline resources, as a lot of it was put in place before that feature existed).

**Comment by shortdudey123** [comment]
> using use_inline_resources will break compatibility with older versions of chef

Can you elaborate?

**Comment by brianbianco** [comment]
Of the top of my head I can't recall the version, but this feature did not exist in older versions of chef (10 series?).  Back then you had no way of propagating changes in resources nested inside other resources.  Redisio implemented a helper feature that would handle this so that it could do that.  In fact I'm pretty sure the use_inline_resource stuff is directly based on that code.  Either way, if you are running that older version of chef (damned if i remember what version) you can't use "use_inline_resource".  The cookbook will probably break as that doesn't even exist.

**Comment by martinb3** [comment]
> i don't see any notifies in the providers which is what would break.

I'm thinking of any third party or downstream cookbook that uses this one. Anyone relying on a notification, when this change ships, would be broken. I'm not opposed to it -- just pointing out it's probably a breaking change.

**Comment by shortdudey123** [comment]
> but this feature did not exist in older versions of chef
I have it guarded with `if defined?(use_inline_resources)`
Or are you saying that the version you are thinking of was after `use_inline_resources` was added?
I am adding kitchen testing for chef versions `11.0.0` and `latest`

> I'm thinking of any third party or downstream cookbook that uses this one
Gotcha, would you rather i not introduce breaking changes in this PR? I could create a separate one that can be merged at a later time

**Comment by brianbianco** [comment]
Oh great, sorry I totally forgot about this and should have reviewed things more closely.  I'm all for moving the breaking change inot a different PR and merging this one.

**Comment by shortdudey123** [comment]
> moving the breaking change inot a different PR and merging this one.

Will do

**Comment by shortdudey123** [comment]
Updated and removed `use_inline_resources` with a `FC059` ignore added for now.

**Comment by martinb3** [comment]
:shipit: 👍

**Pull request merged**